src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/InMemoryEventStore.java

```diff
@@ -6,81 +6,27 @@
 import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.List;
-import java.util.concurrent.atomic.AtomicInteger;
 
 public class InMemoryEventStore implements EventStore {
-    private static final AtomicInteger BUFFER_CONSUMER_THREAD_NUMBER = new AtomicInteger(1);
-    private static final String BUFFER_CONSUMER_THREAD_NAME_PREFIX = "snowplow-eventStore-BufferConsumer-thread-";
-
-    // Thread for moving events from initial buffer into staging buffer
-    private final Thread bufferConsumer;
-    // Queue for immediate buffering of events
-    public final BlockingQueue<TrackerEvent> eventInitialBuffer = new LinkedBlockingQueue<>();
-    // Queue for storing events until bufferSize is reached
-    public final BlockingQueue<TrackerEvent> eventStagingBuffer = new LinkedBlockingQueue<>();
-
-    protected InMemoryEventStore() {
-        bufferConsumer = new Thread(
-                getBufferConsumerRunnable(),
-                BUFFER_CONSUMER_THREAD_NAME_PREFIX + BUFFER_CONSUMER_THREAD_NUMBER.getAndIncrement()
-        );
-        bufferConsumer.start();
-    }
-
-    @Override
-    public List<TrackerEvent> retrieveAllEvents() {
-        return new ArrayList<>(eventStagingBuffer);
-    }
+    public final BlockingQueue<TrackerEvent> eventBuffer = new LinkedBlockingQueue<>();
 
     @Override
     public boolean add(TrackerEvent trackerEvent) {
-        return eventInitialBuffer.offer(trackerEvent);
+        return eventBuffer.offer(trackerEvent);
```

Comment:
It's not clear if LinkedBlockingQueue works correctly in multithreading.
Maybe to avoid issues we could _synchronize_ the methods of `InMemoryEventStore` so we are sure that operations on the queue aren't executed by multiple threads at the same time.

Reply:
LinkedBlockingQueue is thread safe, its why I selected it for the implementation. You shouldn't need to synchronise on the `offer` method.

Here's a better explanation than I can give: https://stackoverflow.com/questions/26543807/is-blockingqueue-completely-thread-safe-in-java